1.19_linux/runtime/cgo/hitsumabushi_syscalls_linux.c

```diff
@@ -3,109 +3,19 @@
 
 // This file defines C functions and system calls for Cgo.
 
-#include <pthread.h>
-#include <errno.h>
-#include <string.h>
-#include <signal.h>
-#include <stdlib.h>
-#include <stdatomic.h>
+#include <pthread.h> // for pthread_self
+#include <stdint.h> // for uint32_t etc.
+#include <stdlib.h> // for exit()
 #include <unistd.h> // for usleep
 #include <stddef.h> // for size_t
-#include <signal.h> // for sigset_t and struct sigaction
+#include <time.h> // for struct timespec
 
 #include "libcgo.h"
 #include "libcgo_unix.h"
 
-typedef unsigned int gid_t;
-
 extern int hitsumabushi_clock_gettime(clockid_t clk_id, struct timespec *tp);
 extern int32_t hitsumabushi_getproccount();
 
-void *mmap(void *addr, size_t length, int prot, int flags, int fd, off_t offset) {
```

Comment:
Is this really safe to remove pseudo function definitions? If so, can we remove them first in another PR as a pure refactoring?

Reply:
My guess is that the compiler succeeds to compile the library as a c-achive, but linking them fails for the final binary.

Reply:
It’s only safe once the rest of the PR is applied, that’s why it’s the last commit. Do you want me to remove that commit and submit it separately?

Reply:
Hm, I don't understand why we can remove these function definitions now...

Reply:
Because all the Go code that uses these functions is disabled by the first commits of this PR.

Reply:
https://github.com/hajimehoshi/hitsumabushi/pull/21/commits/3d8ba41453977a36318c8682cdf6db8f3e3349b3

Did this remove mmap calls?